### PR TITLE
fix: more accurate validation on geneset name

### PR DESCRIPTION
**Pull request by seve**
#### Reviewers
**Functional:** @tihuan @colinmegill 

---

## Changes

- modify
  -  rewrite parts of the validation code to be more accurate, also change some copy

https://user-images.githubusercontent.com/8716829/133507093-d63501bd-c2cb-4811-bf52-737de9c5fdc8.mov

#2414

### CODE REVIEW

File: client/src/components/annoDialog.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ class AnnoDialog extends React.PureComponent {
               <p
                 style={{
                   marginTop: 7,
-                  visibility: validationError ? "visible" : "hidden",
```
Comment (seve) [inline client/src/components/annoDialog.js:45]: validation error doesn't necessarily mean we'll be showing a message

**Comment by signechambers1** [comment]
LGTM! Great work, thanks Seve!

File: client/src/components/annoDialog.js (hunk 1 above)
Comment (tihuan) [reply client/src/components/annoDialog.js:45]: Maybe add this comment in the code just in case someone changes it back to `validationError` lol

**Review by tihuan — APPROVED** [review]
LGTM! Thanks for improving this, Seve 🎉 🥇

**Pull request merged**